database/src/main/java/com/firebase/ui/database/FirebaseIndexRecyclerAdapter.java

```diff
@@ -59,8 +59,11 @@
  * </pre>
  *
  * @param <T>  The Java class that maps to the type of objects stored in the Firebase location.
- * @param <VH> The ViewHolder class that contains the Views in the layout that is shown for each object.
+ * @param <VH> The ViewHolder class that contains the Views in the layout that is shown for each
+ *             object.
+ * @deprecated use {@link com.firebase.ui.database.adapter.FirebaseIndexRecyclerAdapter} instead
  */
+@Deprecated
```

Comment:
@samtstern What do you think of doing a "Screw it all, let's release v2" type of thing? People would just have to make their overridden methods public which doesn't seem so bad. If not, I like your solution, but the method name changes would kill me:

- `onChildChanged` -> `onChangedEvent` I actually kinda like this one.
- `onCancelled` -> `onDatabaseError` I think `onError` would be nicer
- `onDataChanged` - `onNewData` This one is just a blatant lie because it will get called for data changes, removals permutations, etc.
- `parseSnapshot` -> `parseDataSnapshot` This just makes me 😢

If we decided to go the method rename route, `onNewData` absolutely needs to be fixed. Other than that, I guess it's okay. Like you said, it wouldn't be ideal but it's better than being like "Screw y'all, everything you're using is deprecated!" 😄